braintrust/trace/attachment/attachment.go

```diff
@@ -0,0 +1,154 @@
+// Package attachment provides utilities for creating and managing attachments
+// in Braintrust traces.
+//
+// Attachments allow you to log arbitrary binary data, like images, audio, video,
+// PDFs, and large JSON objects, as part of your traces. This enables multimodal
+// evaluations, handling substantial data structures, and advanced use cases such
+// as summarizing visual content or analyzing document metadata. Attachments are
+// securely stored in an object store and associated with your organization.
+//
+// Most users won't need to use this package directly, as the instrumentation
+// middleware (traceopenai, traceanthropic) automatically handles attachment
+// conversion. This package is primarily useful for:
+//   - Manual logging without instrumentation
+//   - Custom scenarios not covered by auto-instrumentation
+//   - Writing instrumentation for new providers
+//
+// For more information, see: https://www.braintrust.dev/docs/guides/attachments
+package attachment
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+)
+
+// Common MIME types for attachments
+const (
+	ImageJPEG = "image/jpeg"
+	ImagePNG  = "image/png"
+	ImageGIF  = "image/gif"
+	ImageWEBP = "image/webp"
+	TextPlain = "text/plain"
+	PDF       = "application/pdf"
+)
+
+// Attachment represents a file or binary data attachment in Braintrust format.
+// When serialized to JSON, it produces the format expected by Braintrust:
+//
+//	{"type": "base64_attachment", "content": "data:image/jpeg;base64,..."}
+type Attachment struct {
```

Comment:
maybe to minimize confusion we call this a `Base64Attachment`?

the `Attachment` types in py/ts are stored in s3 and serialize with type `braintrust_attachment`. this base64 content will be turned into a native attachment and down the line we probably want to have native attachments in the go sdk as well

Reply:
what do you want the api to be in the long run? how will people use each